balena.sh

```diff
@@ -36,10 +41,15 @@ function curl_with_opts() {
 
 function config_from_metadata() {
     #shellcheck disable=SC2034,SC2039 # /bin/sh is a symbolic link to bash on balenaOS
-    for url in "${metadata_urls[@]}"; do
-        user_data="$(curl_with_opts "${url}")"
+    for metadata_url in "${metadata_urls[@]}"; do
+        url="$(echo "${metadata_url}" | awk -F';' '{print $1}')"
+        headers="$(echo "${metadata_url}" | awk -F';' '{print $2}')"
+        user_data="$(curl_with_opts ${headers} "${url}")"
+        if echo "${user_data}" | base64 -d; then
+            user_data="$(echo "${user_data}" | base64 -d)"
+        fi
 
-        if [ -n "${user_data}" ]; then
+        if [ -n "${user_data}" ] && echo "${user_data}" | jq -r .; then
```

Comment:
Is the `echo "${user_data}" | jq -r .` trying to check the validity of the json? Maybe `jq -e . < "${user_data}" works best, because without `-e` jq will exist with success if it runs without compile errors, so a return of `null` will also be successful.

Reply:
@alexgg yes, checking for valid json, user_data is an env-var, `jq -e . < "${user_data}"` requires a file?

Reply:
Indeed, not sure what `${user_data}` contained. Anyway, I was referring more to the `-e` than the echo.

Reply:
@alexgg can you please commit your suggestion (if you still have one).

Reply:
@ab77 I will. My point was that if "${user_data}" is null, `jq -r` will output success while `jq -e` will output error.
```
$ cat null.json 
null
$ cat null.json | jq -r .
null
$ echo $?
0
$ cat null.json | jq -e .
null
$ echo $?
1
```

Reply:
Oh, I see - I'll test this out and incorporate next week.